### DISCUSSION TITLE
Support for unstructured grids

### DISCUSSION THREAD

**Discussion by imab4bsh** [Parcels-code/Parcels#1006] [Q&A]
Dear Ocean Parcels team,
In your website, under future development goals there is the topic of "Support for unstructured grids". I wonder if there is already people working on this feature since I know of some people in Germany that would gladly collaborate in this effort. 
Best, Ingrid
ps. I hope this is the right place to ask this question!

**Comment by erikvansebille** [comment]
Hi @imab4bsh, thanks for bringing this up! We had a couple of chats over the last few months with @465b, who was also interested in unstructured grid support in Parcels. This could be a good thread to discuss that further. When you say 'collaborate', do you mean helping with development?

By the way, a few weeks ago [Ross Vennell et al](https://link.springer.com/article/10.1007%2Fs10236-020-01436-7) published a very relevant paper in Ocean Dynamics , with an implementation for Lagrangian particle tracking on unstructured grids. That could be an interesting starting point for anyone wanting to implement this in Parcels too?

**Reply by koldunovn** [reply]
Hi @erikvansebille . We in particular interested in including FESOM2 (https://github.com/FESOM/fesom2) to parcels. Some Lagrangian code for FESOM already exist, and the easiest route, I guess, would be first to port it to parcels. 

Could you do an approximate estimate of how hard it could be, given the current architecture of parcels?

The ideal case, I guess, is if someone from our side try to implement it under parcel developers guidance, but it would help to find resources if we know, at least very approximately, how hard it could be and how long it might take :)

**Reply by willirath** [reply]
Slightly off topic: Including more general grids could be a great opportunity for moving towards arbitrary indexers (see parts of https://github.com/OceanParcels/parcels/issues/462). I have done some experiments with [xoak](https://xoak.readthedocs.io) in [this gist](https://gist.github.com/willirath/bdcdb779080febd3d374d9cdfa769378) (which are in no way a replacement for Parcels, but could show the way towards stricter modularization of the different components of parcels).

**Comment by backeb** [comment]
@erikvansebille interested to know the status of unstructured grid support. Has there been any further development?
Cheers
Björn

**Reply by erikvansebille** [reply]
No, there hasn't been much update in the last year as far as I know. We don't work with unstructured grid-models in my own group; so have no experience. I don't know of others actually developing this

**Reply by 465b** [reply]
@backeb @erikvansebille 
Ross Vennell just put the new version of oceantracker (an unstructured grid lagrangian tracker) on github.
(https://github.com/oceantracker/oceantracker)
Might be worth checking out as it offers some of the functionality of parcels
